src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -257,8 +258,331 @@ private struct IntPtrAndHandle
+        public static unsafe string GetCustomMarshalerInfoForType (IntPtr typePtr, string marshalerFullName) {
+            if (typePtr == IntPtr.Zero)
+                return "null";
+            else if (string.IsNullOrEmpty(marshalerFullName))
+                return "null";
+
+            IntPtrAndHandle tmp = default(IntPtrAndHandle);
+            tmp.ptr = typePtr;
+            var typeHandle = tmp.typeHandle;
+
+            var type = Type.GetTypeFromHandle(typeHandle);
+            var marshalerType = Type.GetType(marshalerFullName) ?? type.Assembly.GetType(marshalerFullName);
+            if (marshalerType == null)
+                return "null";
+
+            var preFilter = GetAndEscapeStringProperty(marshalerType, "FromJavaScriptPreFilter");
+            var postFilter = GetAndEscapeStringProperty(marshalerType, "ToJavaScriptPostFilter");
+
+            var inputPtr = GetMarshalMethodPointer(marshalerType, "FromJavaScript", out Type? fromReturnType, out Type? fromParameterType);
+            var outputPtr = GetMarshalMethodPointer(marshalerType, "ToJavaScript", out Type? toReturnType, out Type? toParameterType);
+
+            if (inputPtr == IntPtr.Zero)
+                throw new Exception($"{marshalerType.Name} must have a static FromJavaScript method");
+            if (outputPtr == IntPtr.Zero)
+                throw new Exception($"{marshalerType.Name} must have a static ToJavaScript method");
+
+            if (fromReturnType != type)
+                throw new Exception($"{marshalerType.Name}.FromJavaScript's return type must be {type.Name} but was {fromReturnType}");
+
+            if (type.IsValueType) {
+                if ((toParameterType == null) || !toParameterType.IsByRef || (toParameterType.GetElementType() != type))
+                    throw new Exception($"{marshalerType.Name}.ToJavaScript's parameter must be 'ref {type.Name}' but was {toParameterType}");
+            } else {
+                if (toParameterType != type)
+                    throw new Exception($"{marshalerType.Name}.ToJavaScript's parameter must be of type {type.Name} but was {toParameterType}");
+            }
+
+            return ("{\n" + $"\"typePtr\": {typePtr}, \n" +
+                $"\"preFilter\": {preFilter}, \"postFilter\": {postFilter}, \n" +
+                $"\"inputPtr\": {inputPtr}, \"outputPtr\": {outputPtr} \n" +
+                "}");
+        }
+
+        public static MarshalType GetMarshalTypeFromType (Type type) {
+            if (type == null)
+                return MarshalType.VOID;
+
+            MarshalType? result = null;
+            switch (Type.GetTypeCode(type)) {
+                case TypeCode.Byte:
+                case TypeCode.SByte:
+                case TypeCode.Int16:
+                case TypeCode.UInt16:
+                case TypeCode.Int32:
+                    result = MarshalType.INT;
```

Comment:
Why not simply `return MarshalType.INT;` ?

Reply:
I recall having a reason for this, but it might not be valid anymore. I'll clean this method up